src/main/java/Tasks/Task.java

```diff
@@ -0,0 +1,78 @@
+package Tasks;
+
+/**
+ * Represents a generic task.
+ */
+public class Task {
+
+    protected String name;
+    protected boolean completed;
+
+    /**
+     * Constructs a Task object with the specified name.
+     *
+     * @param name the name of the task
+     */
+    public Task(String name) {
+        this.name = name;
+        this.completed = false;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void markTask() {
+        this.completed = true;
+    }
+
+    /**
+     * Sets the task as completed.
+     */
+    public void setCompleted() {
```

Comment:
the setCompleted() and unmarkTask() could be standardised and have similar names since their functions are similar